spec/controllers/campaigns_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe CampaignsController, type: :controller do
+  let(:campaign) { create(:campaign) }
+
+  describe 'GET #index' do
+    it 'renders index template' do
+      get :index
+      is_expected.to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
```

Comment:
Add the test case when the ID is invalid

Reply:
added